### DISCUSSION TITLE
component is deprecated.

### DISCUSSION THREAD

**Discussion by ghost** [slint-ui/slint#5383] [Q&A] [answered]
if i take this as a code template 
```
import { StandardButton, GridBox, LineEdit } from "std-widgets.slint";
_ := Dialog {
    property name <=> name-le.text;
    property address <=> address-le.text;
    StandardButton { kind: ok; }
    StandardButton { kind: cancel; }
    preferred-width: 300px;
    GridBox {
        Row {
            Text { text: "Enter your name:"; }
            name-le := LineEdit { }
        }
        Row {
            Text { text: "Address:"; }
            address-le := LineEdit { }
        }
    }
}
```
then the declarable gives 
```
':=' to declare a component is deprecated. The new syntax declare components with 'component MyComponent {'
```
any solution for this.

**Comment by hunger** [comment]
I would opt for a more descriptive name than `_` like maybe `UserDataDialog`:

```slint
import { StandardButton, GridBox, LineEdit } from "std-widgets.slint";
component UserDataDialog inherits Dialog {
    property name <=> name-le.text;
    property address <=> address-le.text;

    StandardButton { kind: ok; }
    StandardButton { kind: cancel; }

    preferred-width: 300px;

    GridBox {
        Row {
            Text { text: "Enter your name:"; }
            name-le := LineEdit { }
        }
        Row {
            Text { text: "Address:"; }
            address-le := LineEdit { }
        }
    }
}
```

Basically `X := Foo { }` becomes `component X inherits Foo {}`, but actually `component X { }` works now, too.

You might also want to `export` the components you want to access from your business logic or from other .slint files, which would turn the above into `export component UserDataDialog inherits Dialog {`.

I hope this helps!

**Comment by ghost** [comment]
@hunger thank you and would it be possible to export the variable that we are reading in Text to another variable by shadowing the variable or implementing the hostig. Let me know. 

I get to learn the SlintUI first through shell and then putting the javascript and rust later on for some of the applications development. Thank you, Gaurav

**Comment by ghost** [comment]
@hunger thank you and here is a piece of the code that i am putting into the SlintUI. Basically, the conda package search doesnt allow you to code with the DOM, as the page is protected, so i implemented this to fetch the DOM and then alter the terms internally otherwise would have used the Cheerio through the DOM manipulation and eventListener. 
```
#!/usr/bin/bash
# Author Gaurav
# Universitat Potsdam
# Date 2024-6-6
# A way to connect to the conda and search for the terms and get the conda generators.
term=$(slint-viewer - --save-data - << EOF
import { StandardButton, GridBox, LineEdit } from "std-widgets.slint";
component UserDataDialog inherits Dialog {
    property name <=> name-le.condasearch;
    StandardButton { kind: ok; }
    StandardButton { kind: cancel; }
    preferred-width: 300px;
    GridBox {
        Row {
            Text { text: "Enter your conda search:"; }
            name-le := LineEdit { }
        }
        }
    }
}
EOF
)
wget -F "https://anaconda.org/search?q=${term}" -O condasearch.txt
echo "all term fetched"
cat condasearch.txt | grep ${term} | \
         grep href | cut -f 2 -d "=" \
              | cut -f 1 -d ">" | grep -w ${term} > saveintermediate.txt
echo "all intermediate generated"
for i in $(cat saveintermediate.txt); do echo \
        "https@//anaconda.org$i"; done | \
                     sed "s/\"//g" | sed "s/@/:/g" > downloadlinks.txt
echo "moving into intermediate"
mkdir intermediate
cp -r downloadlinks.txt ./intermediate
cd intermediate
cat downloadlinks.txt | while read line; \
          do wget -F ${line}; done
cat * | grep -w install | grep ${term} > newsave.txt
cat newsave.txt | grep code | cut -f 2 -d ">" | cut -f 1 -d "<"
cd ..
rm -rf intermediate
rm -rf saveintermediate.txt finaltext.txt downloadlinks.txt condasearch.txt
```
Now the point of question is that do i need to provide the term that i am getting through the slint output as <<<{term} or it will store the variable and will listen as ${term}. 

Thank you for the explanation,
Gaurav

**Comment by ogoffart** [comment]
Thanks for your question, I now realize this code is coming from our repository.
It was still using the old syntax (pre-1.0)
I have updated the example in https://github.com/slint-ui/slint/commit/9f6db8977001fd4997fcebd08807fd5ee7eb2b3b

In short, we now use `component` to declare a component, and the properties needs to be annotated with `in-out` otherwise they are private.

**Reply by ghost** [reply]
@hunger thank you and yes i looked at the example files to understand the language syntax and now have learnt how the slint language is coded. Now the question that i have is: 

so if i describe a out variable then it will be used as a inline variable and then can be used for the downstream. is it so?. 

Thank you,
Gaurav

**Reply by ogoffart** [reply]
I'm not sure i understand the question. If you define the property as `out` it will be in the json written by the `--save-data`

**Reply by ghost** [reply]
ok, i understood now. I will also add a PopupWindow and will update the code. Thank you Olivier.

**Reply by ghost** [reply]
@ogoffart no this is what i was expecting in your example files in the slint codebase, it is giving a redirection error. 
```
sh simple_input.sh
Syntax error: redirection unexpected
```
i have the cargo and both the slint-lsp installed.

**Reply by ogoffart** [reply]
The code example uses some bash extension and don't work with just `sh`.  Try `bash simple_input.sh`

**Reply by ghost** [reply, marked as answer]
thank you and i didnt realized that it will not pick up the zsh shell.
